service/kernel/data_management/train_test_split.cpp

```diff
@@ -0,0 +1,256 @@
+    {
+        const size_t nTestBlocks = daal::services::internal::max<cpu, size_t>(nTestRows / BLOCK_SIZE, 1);
+        const int nTestSurplus = nTestRows - BLOCK_SIZE * nTestBlocks;
+
+        daal::threader_for(nTestBlocks, nTestBlocks, [&](size_t iBlock) {
+            const size_t start = iBlock * BLOCK_SIZE;
+            const size_t end   = iBlock == nTestBlocks - 1 ? start + BLOCK_SIZE + nTestSurplus : start + BLOCK_SIZE;
+
+            s |= assignRows<DataType, IdxType, cpu>(origDataPtr, testTable, testIdxTable, start, end - start, nColumns);
+        });
+    }
+    else
+    {
+        s |= assignRows<DataType, IdxType, cpu>(origDataPtr, testTable, testIdxTable, 0, nTestRows, nColumns);
+    }
+
+    return s;
+}
+
+template <typename IdxType, daal::CpuType cpu>
+services::Status trainTestSplitImpl(NumericTable & inputTable, NumericTable & trainTable, NumericTable & testTable,
+                                    NumericTable & trainIdxTable, NumericTable & testIdxTable)
+{
+    services::Status s;
+    const size_t nThreads = threader_get_threads_number();
+    const NTLayout layout  = inputTable.getDataLayout();
+    const size_t nColumns = trainTable.getNumberOfColumns();
+    const size_t nTrainRows = trainTable.getNumberOfRows();
+    const size_t nTestRows = testTable.getNumberOfRows();
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nTrainRows + nTestRows, nColumns);
+
+    NumericTableDictionaryPtr tableFeaturesDict = inputTable.getDictionarySharedPtr();
+
+    if (layout == NTLayout::soa)
+    {
+        BlockDescriptor<IdxType> trainIdxBlock, testIdxBlock;
+        trainIdxTable.getBlockOfColumnValues(0, 0, nTrainRows, readOnly, trainIdxBlock);
+        testIdxTable.getBlockOfColumnValues(0, 0, nTestRows, readOnly, testIdxBlock);
+        const IdxType * trainIdx = trainIdxBlock.getBlockPtr();
+        const IdxType * testIdx = testIdxBlock.getBlockPtr();
+        DAAL_CHECK_MALLOC(trainIdx);
+        DAAL_CHECK_MALLOC(testIdx);
+
+        runBlocks(nColumns > 1 && nColumns * (nTrainRows + nTestRows) > THREADING_BORDER && nThreads > 1, nColumns, [&](size_t iCol) {
+            switch ((*tableFeaturesDict)[iCol].getIndexType())
+            {
+                case daal::data_management::features::IndexNumType::DAAL_INT32_S:
+                    s |= splitColumn<int, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+                    break;
+                case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+                    s |= splitColumn<float, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+                    break;
+                case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+                    s |= splitColumn<double, IdxType, cpu>(inputTable, trainTable, testTable, trainIdx, testIdx, nTrainRows, nTestRows, iCol, nThreads);
+                    break;
```

Comment:
We have following possible list of types for IndexType:
```cpp
enum IndexNumType
{
    DAAL_FLOAT32 = 0,
    DAAL_FLOAT64 = 1,
    DAAL_INT32_S = 2,
    DAAL_INT32_U = 3,
    DAAL_INT64_S = 4,
    DAAL_INT64_U = 5,
    DAAL_INT8_S  = 6,
    DAAL_INT8_U  = 7,
    DAAL_INT16_S = 8,
    DAAL_INT16_U = 9,
    DAAL_OTHER_T = 10
};
```
Currently you handle only 3 types. Is it expected?

Reply:
All types support is added